### PR TITLE
Fixed Swift Package Manager Errors for v2

### PR DESCRIPTION
**Pull request by Sam-Spencer**
When integrating v2 via SPM, the current v2 branch has build errors that prevent any project from compiling. I've gone ahead and updated the deployment target for SPM and added some missing import statements. Now things should compile correctly.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -4,7 +4,7 @@ import PackageDescription
 let package = Package(
   name: "Pixel",  
   platforms: [
-    .iOS(.v12)
+    .iOS(.v14)
```
Comment (muukii) [inline Package.swift:7]: Can we keep v12?

**Comment by muukii** [comment]
@Sam-Spencer 
Wow, thank you for a great contribution.
I put a comment, please check it.

**Comment by muukii** [comment]
oh, sorry. it's conflicted.
My other working branch broke this PR...

File: Package.swift (hunk 1 above)
Comment (Sam-Spencer) [reply Package.swift:7]: It needs to be moved to at least iOS 13. It won't compile from SPM on iOS 12 because some of the dependencies use 13+ APIs.

**Comment by Sam-Spencer** [comment]
Fixed the conflicts! Things should work now :)

File: Package.swift (hunk 1 above)
Comment (muukii) [reply Package.swift:7]: hm, okay.
I'll check it later

**Comment by muukii** [comment]
@Sam-Spencer 
I got errors:
```
Fetching https://github.com/muukii/TransitionPatch.git
Fetching https://github.com/ReactiveX/RxSwift.git
Fetching https://github.com/VergeGroup/Verge.git
Cloning https://github.com/VergeGroup/Verge.git
Resolving https://github.com/VergeGroup/Verge.git at 8.8.0
Cloning https://github.com/ReactiveX/RxSwift.git
Resolving https://github.com/ReactiveX/RxSwift.git at 6.1.0
Cloning https://github.com/muukii/TransitionPatch.git
Resolving https://github.com/muukii/TransitionPatch.git at 1.0.3
error: the library 'PixelEngine' requires macos 10.10, but depends on the product 'Verge' which requires macos 10.12; consider changing the library 'PixelEngine' to require macos 10.12 or later, or the product 'Verge' to require macos 10.10 or earlier.
error: the library 'PixelEditor' requires macos 10.10, but depends on the product 'Verge' which requires macos 10.12; consider changing the library 'PixelEditor' to require macos 10.12 or later, or the product 'Verge' to require macos 10.10 or earlier.
```

It seems SwiftPM tried to build for macos.
How do we only build for iOS?

**Pull request merged**